src/util/request-manager.js

```diff
@@ -412,7 +414,16 @@ export default class RequestManager {
     if (params.url.startsWith('https:')) {
       proxy = this.httpsProxy;
     }
-    params.proxy = String(proxy);
+
+    if (proxy) {
+      // if no proxy is set, do not pass a proxy down to request.
+      // the request library will internally check the HTTP_PROXY and HTTPS_PROXY env vars.
+      params.proxy = String(proxy);
+    } else if (proxy === false) {
+      // passing empty string prevents the underlying library from falling back to the env vars.
+      // an explicit false in the yarn config should override the env var. See #4546.
+      params.proxy = '';
+    }
```

Comment:
what happens to other falsey variables like 0 or undefined? Maybe you should flip the order?
```
if (proxy === false) {
   params.proxy = '';
} else {
   params.proxy = String(proxy);
}
``` 
or if they're meant to be empty strings
```
params.proxy = proxy ? String(proxy) || '';
```

Reply:
Based on some logic above:
```
    if (opts.httpsProxy === '') {
      this.httpsProxy = opts.httpProxy || '';
    } else if (opts.httpsProxy === false) {
      this.httpsProxy = false;
    } else {
      this.httpsProxy = opts.httpsProxy || '';
    }
```

so `httpsProxy` should be one of `''`, `false`, `'https://wherever`

In the case of `''` we don't want to set any `params.proxy` at all because the `request` library will check `hasOwnProperty('httpsProxy')` on the object we pass it. If that property exists at all (even if set to a falsey value) then it will not check the env vars.

So basically the logic here is:

```
if ''
  don't set a proxy (request will fall back to env vars)
if false
  set a proxy to '' (request will not use env vars, and not use any proxy)
if 'some non-falsy string'
  set a proxy to that string (request will use this as the proxy)
```

Hopefully that makes sense...

Reply:
I see. Thanks for explaining!